### PR TITLE
Handling User Authentication in Firebase and Vue.js Applications

### LINKED ISSUE #5452: Managing User State in Vue and Firebase Applications
Vuex is a state management library for Vue applications. State management in Vue applications is important as it enables any component within the application to directly access and mutate the state as required. We will take a look at how we can use Vuex 4 and the composition API to set up a state management system for authenticating users with firebase v9 auth. With the state management system in place, we will be able to access the logged-in state from any component and take the necessary actions.

Some of the topics that will be covered in this article include:
- Basics of Vuex and the Composition API.
- Setting up Firebase Auth.
- Setting up an auth store.
- Signing up users.
- Logging in users.
- Logging out users.

#### Key takeaways 
- Working with the Composition API and Vuex 4.
- Managing user state in Vue applications.
- Working with Vue and Firebase.

#### Article quality
The article will be beginner-friendly as it will guide the user on how to achieve the above-stated goals from scratch. Also, articles published on the web that cover this topic use the older versions of Vue, Vuex and, Firebase. We will cover this article using the Vue 3.0, Vuex 4, and Firebase v9, which are all the latest versions. The other articles also use the Options API but in this article, we are going to use the composition API.

#### References
N/A

### Templates to use as guides
N/A

### PR DESCRIPTION
**Pull request by annemwangi**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5452

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by srishilesh** [comment]
Hello @annemwangi ,

Thanks for submitting this PR.

Your PR has entered our reviewing queue. Please wait until a peer-reviewer pings you with the changes to be made.

Meanwhile, we suggest you to go through other approved PRs/published articles to see if you have missed out anything, which might avoid a potential PR closure.

Thanks.

**Comment by adrianmurage** [comment]
@section-engineering-education/content-moderator-team

**Comment by louisefindlay23** [comment]
Plagiarism check complete

**Review by adrianmurage — CHANGES_REQUESTED** [review]
@annemwangi 
Your tutorial reads smoothly but, sadly, in the end nothing renders.
Meaning that you will need to review your article and make sure it's working as expected.

The best way to do this is to start from scratch and follow your tutorial word for word and see if you end up with the expected end result. This will help you catch anything you might have omitted in the writing process.

I'll be closing this PRuntil you do. Once everything is ready we can reopen the PR.

@ me when you're ready for me to take a look

**Comment by annemwangi** [comment]
@adrianmurage, could you please explain to me what "nothing renders means"?
Because the code was working perfectly. I have also tested it right now and it's still working perfectly.

**Comment by adrianmurage** [comment]
@annemwangi 

after following through your article with 2 different machines, I end up with the same issue:
```bash
 ERROR  Failed to compile with 1 error                                                                                         1:58:59 PM


/home/m/Code/reviews/firebase-vue-user-management/src/components/Navbar.vue
  1:1  error  Component name "Navbar" should always be multi-word  vue/multi-word-component-names

/home/m/Code/reviews/firebase-vue-user-management/src/views/Home.vue
  1:1  error  Component name "Home" should always be multi-word  vue/multi-word-component-names

/home/m/Code/reviews/firebase-vue-user-management/src/views/Login.vue
  1:1  error  Component name "Login" should always be multi-word  vue/multi-word-component-names

/home/m/Code/reviews/firebase-vue-user-management/src/views/Signup.vue
  1:1  error  Component name "Signup" should always be multi-word  vue/multi-word-component-names

✖ 4 problems (4 errors, 0 warnings)


You may use special comments to disable some warnings.
Use // eslint-disable-next-line to ignore the next line.
Use /* eslint-disable */ to ignore all warnings in a file.
ERROR in 
/home/m/Code/reviews/firebase-vue-user-management/src/components/Navbar.vue
  1:1  error  Component name "Navbar" should always be multi-word  vue/multi-word-component-names

/home/m/Code/reviews/firebase-vue-user-management/src/views/Home.vue
  1:1  error  Component name "Home" should always be multi-word  vue/multi-word-component-names

/home/m/Code/reviews/firebase-vue-user-management/src/views/Login.vue
  1:1  error  Component name "Login" should always be multi-word  vue/multi-word-component-names

/home/m/Code/reviews/firebase-vue-user-management/src/views/Signup.vue
  1:1  error  Component name "Signup" should always be multi-word  vue/multi-word-component-names

✖ 4 problems (4 errors, 0 warnings)


webpack compiled with 1 error
```

Screenshot from 2022-02-23 13-59-33

**Comment by adrianmurage** [comment]
I started again from scratch up to the point where you introduce Navbar.
Upon renaming `Navbar.vue` to `NavBar.vue`, I got it to render out.

I also found [documentation](https://v2.vuejs.org/v2/style-guide/#Multi-word-component-names-essential) that explains the error. 

I am assuming if you fix this, the app should work as expected

Screenshot from 2022-02-23 14-50-12
Screenshot from 2022-02-23 14-50-23

File: content/articles/firebase-vue-authentication/index.md [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+Open the terminal and type:
+```bash
+npm install firebase
+```
+This will install firebase.
+
+After the installation is complete, create a new folder in the `src` folder and name it `firebase`. Inside this folder, create a file `config.js`. Modify the file with the following code:
+```js
+//Import the required methods
+import { inititializeApp } from 'firebase/app'
```
Comment (adrianmurage) [inline content/articles/firebase-vue-authentication/index.md:70]: The word `initialize` is spelled wrong and that is what was causing my app to not render.

```suggestion
import { initializeApp } from 'firebase/app'
```

File: content/articles/firebase-vue-authentication/index.md [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+    apiKey: "",
+    authDomain: "",
+    projectId: "",
+    storageBucket: "",
+    messagingSenderId: "",
+    appId: ""
+  };
+
+//initialize the firebase app
+inititializeApp(firebaseConfig)
```
Comment (adrianmurage) [inline content/articles/firebase-vue-authentication/index.md:84]: The word `initialize` is spelled wrong and that is what was causing my app to not render.

```suggestion
initializeApp(firebaseConfig)
```

**Review by adrianmurage — COMMENTED** [review]
@annemwangi 

It seems that the reason my app was not rendering was as a result of a typo in your `firebase/config.js` file

**Comment by adrianmurage** [comment]
@annemwangi 
Good job with this article.

Do however take note of the value of working through your article from a reader's perspective to help catch any errors in your future work

Otherwise, keep up the good work and I look forward to seeing more of your work published by EngEd

**Comment by annemwangi** [comment]
> @annemwangi Good job with this article.
> 
> Do however take note of the value of working through your article from a reader's perspective to help catch any errors in your future work.
> 
> Otherwise, keep up the good work and I look forward to seeing more of your work published by EngEd.

Thanks a lot for taking the time to review my article and for the advice :rocket:

**Comment by WanjaMIKE** [comment]
Merged #6282 via #5452

**Pull request merged**